pkg/kudoctl/clog/log.go

```diff
@@ -0,0 +1,131 @@
+package clog
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"sync/atomic"
+
+	"github.com/spf13/pflag"
+)
+
+// this package provides (C)lient logs, or CLI logs. Although this is called a log, it is for writing to STDOUT and is designed
+// for CLI output (not to log files etc.). We want -V behavior familiar to kubectl users, but klog comes with a lot of dependencies
+// and global flags that obfuscate KUDO.  glog is built more for a multi-threaded logger (managing buffers etc) which
+// KUDO CLI doesn't have a need for.  clog provides verbosity control for CLI output.
+
+// guidance for use of V level
+//  0-1 normal standard out
+//  2-4 as debug-level logs
+//  5-6 logical chooses
+//  7-8 input/output details
+//  9-10 as trace-level (http details)
+
+// Level specifies a level of verbosity for V logs.
+type Level int32
+
+// get returns the value of the Level.
+func (l *Level) get() Level {
+	return *l
+}
+
+// set sets the value of the Level.
+func (l *Level) set(val Level) {
+	atomic.StoreInt32((*int32)(l), int32(val))
```

Comment:
Overall looks good, but why does this need to be atomic? I'd think with the CLI this would be getting set once, upfront, and so doesn't need to be threadsafe.

Reply:
Also, I'm surprised this typecasting is necessary. Should it be there?

Additional thing, these methods could use Level as a value, not as a reference. Since it's a primitive, I don't see a need for indirection here. Maybe because of copy semantics, worth checking if it takes only a minute or two to test.

Reply:
this was lifted from glog as inspiration.. let me see what we can do here.   We explicit are not looking for thread safety.  Set upfront and go is correct.